test/Conversion/ExportVerilog/instance-choice.mlir

```diff
@@ -0,0 +1,40 @@
+// RUN: circt-opt %s -export-split-verilog='dir-name=%t.dir'
+// RUN: cat %t.dir%{fs-sep}top.sv | FileCheck %s
+
+hw.module private @TargetA(in %a: i32, out b: i32) {
+  hw.output %a : i32
+}
+
+hw.module private @TargetB(in %a: i32, out b: i32) {
+  hw.output %a : i32
+}
+
+hw.module private @TargetDefault(in %a: i32, out b: i32) {
+  hw.output %a : i32
+}
+
+// CHECK-LABEL: module top
+hw.module public @top(in %a: i32, out b: i32, out d: i32) {
+  // CHECK:      `ifndef __circt_choice_top_inst1
+  // CHECK-NEXT: `define __circt_choice_top_inst1 TargetDefault
+  // CHECK-NEXT: `endif
+  // CHECK-NEXT: `__circt_choice_top_inst1 inst1 (
+  // CHECK-NEXT:    .a (a),
+  // CHECK-NEXT:    .b (b)
+  // CHECK-NEXT: );
+  %b = hw.instance_choice "inst1" sym @inst1 @TargetDefault or @TargetA if "A" or @TargetB if "B"(a: %a: i32) -> (b: i32)
```

Comment:
Emission looks great. Restating my comment above that this test is both very clean and well formatted for both understandability and maintainability.